multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py

```diff
@@ -398,12 +398,16 @@ def transform_image(
         assert (
             self._fit_called or self._fit_x_called
         ), "You will need to first call preprocessor.fit_x() before calling preprocessor.transform_image."
+
         image_features = {}
         image_types = {}
         for col_name in self._image_feature_names:
             col_value = df[col_name]
             col_type = self._column_types[col_name]
-            if col_type == IMAGE_PATH or IMAGE:
+
+            if col_name == ROIS:
```

Comment:
Should be `col_type == ROIS`?

Reply:
The column_type here for ROIS column is IMAGE. This hack will be solved in a later PR by using transform_multimodal().

Reply:
In `infer_column_types `, the rois column is detected as type `ROIS`. In lines 211 and 284, the column type is also `ROIS`. It's confusing from where the column type is changed to `IMAGE`.

Reply:
Added a TODO comment here. Per our offline discussion, I'll bypass/overwrite the type fallback in while detection model predicts. This change will be completed in next PR.